mobile/examples/Maui/MauiVisionSample/readme.md

```diff
@@ -0,0 +1,105 @@
+
+
+## Building
+
+To build the sample Visual Studio 2022 Preview is currently required, as .net 6 support (which includes MAUI) is only available in that version.
+
+The MAUI workload should be installed. See [here](https://docs.microsoft.com/en-us/dotnet/maui/get-started/first-app) for more information.
+
+If you get any errors during building about missing platform workloads a simple way to resolve is to open a terminal from Visual Studio 2022 Preview (View->Terminal) and run the command `dotnet workload restore MauiVisionSample.sln`
+
+## ONNX Runtime Nuget Packages
+
+There are two ONNX Runtime NuGet packages used. 
+ - Microsoft.ML.OnnxRuntime
+   - contains the native libraries for all supported platforms, including various architectures of Windows, Linux, Mac, iOS and Android
+ - Microsoft.ML.OnnxRuntime.Managed
+   - contains the C# bindings to use the native libraries
+
+## ONNX Runtime Usage
+
+The key parts demonstrating the integration of ONNX Runtime are:
+
+### Model
+
+The ONNX models are included in the [Raw](MauiVisionSample/Resources/Raw) resources. 
+
+The model bytes are loaded using `OpenAppPackageFileAsync` in `Utils::LoadResource`. Note that it is necessary to copy to a byte[] at runtime as the model may be compressed in the app on platforms such as Android.
+
+### Microsoft.ML.OnnxRuntime.InferenceSession
+
+The `InferenceSession` reads the ONNX model bytes, optimizes the model, and handles model execution.
+
+An `InferenceSession` instance can be used to execute the model multiple times, including concurrently, so should only ever be created once per model. 
+
+Creating an `InferenceSession` that uses the CPU Execution Provider is initially done in `VisionSampleBase::Initialize`.
+
+We also demonstrate the steps to enable an additional execution provider (if available for the platform) in `VisionSampleBase::UpdateExecutionProviderAsync`. 
+As execution providers must be selected prior to the creation of the `InferenceSession` this necessitates an expensive re-creation of the session in the sample app. Typically you would pre-determine which execution providers you wanted enabled and create the session once with that information.
+
+The CPU Execution Provider will be able to run all models.
+
+In some scenarios it may be beneficial to use the NNAPI Execution Provider on Android, or the CoreML Execution Provider on iOS. 
+This is highly dependent on your model, and also the device (particularly for NNAPI), so it is necessary for you to performance test.
+
+## Image acquisition
+
+There are 3 potential ways to acquire the image to process in the sample app.
```

Comment:
> There are 3 potential ways to acquire the image

nit: from info below, seems like there are 2 ways to acquire the image? ( 3 stages for image processing

Reply:
The 3 I'm referring to are using the included sample image (GetSampleImageAsync), picking an existing image from the device (PickPhotoAsync), or taking a photo (TakePhotoAsync).